presto-main/src/main/java/com/facebook/presto/execution/buffer/PagesSerde.java

```diff
@@ -58,20 +56,18 @@ public SerializedPage serialize(Page page)
         writeRawPage(page, serializationBuffer, blockEncodingSerde);
 
         if (!compressor.isPresent()) {
-            return new SerializedPage(serializationBuffer.slice(), UNCOMPRESSED, page.getPositionCount(), serializationBuffer.size());
+            return new SerializedPage(serializationBuffer.slice(), PageCodecMarker.none(), page.getPositionCount(), serializationBuffer.size());
         }
 
-        int maxCompressedLength = maxCompressedLength(serializationBuffer.size());
-        byte[] compressionBuffer = new byte[maxCompressedLength];
-        int actualCompressedLength = compressor.get().compress(serializationBuffer.slice().getBytes(), 0, serializationBuffer.size(), compressionBuffer, 0, maxCompressedLength);
+        byte[] compressionBuffer = new byte[compressor.get().maxCompressedLength(serializationBuffer.size())];
```

Comment:
this change seems unrelated; it is preferable to put unrelated changes into separate commits